src/commands/functions/create.ts

```diff
@@ -1,20 +1,30 @@
+import { SiteNotFoundError } from '@fleek-platform/errors';
 import { output } from '../../cli';
 import type { SdkGuardedFunction } from '../../guards/types';
 import { withGuards } from '../../guards/withGuards';
 import { t } from '../../utils/translation';
 import { getFunctionNameOrPrompt } from './prompts/getFunctionNameOrPrompt';
+import { isSiteIdValid } from './utils/isSiteIdValid';
 
 type CreateFunctionArgs = {
   name?: string;
+  siteId?: string;
 };
 
 const createAction: SdkGuardedFunction<CreateFunctionArgs> = async ({
   args,
   sdk,
 }) => {
-  const functionName = await getFunctionNameOrPrompt({ name: args.name });
+  const { name, siteId } = args;
+  const functionName = await getFunctionNameOrPrompt({ name });
 
-  const newFunction = await sdk.functions().create({ name: functionName });
+  if (siteId && !(await isSiteIdValid({ siteId, sdk }))) {
+    throw new SiteNotFoundError({ site: { id: siteId } });
```

Comment:
Maybe you meant to show an "output" error? Otherwise, if this is not try/out, it'll throw.

```suggestion
    output.error(`Some site not found error`);
```

Reply:
The idea is if you set a siteId, it should be valid, and fail to create the function

Reply:
@gabrielmpinto but when it fails, does it show a nice message or thrown error?